### PR TITLE
Added an assert for FixedPointType in InlineAssembly

### LINKED ISSUE #8412: [solc] ICE in codegen because "variable->type()->category() == Type::Category::Integer" fails
## Description

```
contract C {
  fixed constant b=9;
  function f() public returns (int w) {
    assembly { w := b }
  }
}
```

throws

https://github.com/ethereum/solidity/blob/b65a165da19f150fbff39eac08729da72e4216c3/libsolidity/codegen/ContractCompiler.cpp#L693

## Environment

- Compiler version: latest develop
- Target EVM version (as per compiler settings): NA

## Steps to Reproduce

```
$ solc <input.sol>
Internal compiler error during compilation:
/home/bhargava/work/github/solidity/libsolidity/codegen/ContractCompiler.cpp(693): Throw in function auto solidity::frontend::ContractCompiler::visit(const solidity::frontend::InlineAssembly &)::(anonymous class)::operator()(const yul::Identifier &, yul::IdentifierContext, yul::AbstractAssembly &) const
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] =
```

### PR DESCRIPTION
**Pull request by hrkrshnn**
Closes https://github.com/ethereum/solidity/issues/8412

### CODE REVIEW

**Comment by hrkrshnn** [comment]
The change is too small, so I didn't add a Changelog entry

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -699,6 +699,9 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 				}
 			}
 
+			if (dynamic_cast<FixedPointType const*>(var->type()))
+				solAssert(false, "FixedPointType not implemented.");
```
Comment (mijovic) [inline libsolidity/analysis/TypeChecker.cpp]: Maybe only this?
```suggestion
			solAssert(dynamic_cast<FixedPointType const*>(var->type()), "FixedPointType not implemented.");
```

**Review by mijovic — DISMISSED** [review]
Only small suggestion which is not mandatory, I approve

File: libsolidity/analysis/TypeChecker.cpp (hunk 1 above)
Comment (bshastry) [reply libsolidity/analysis/TypeChecker.cpp]: I guess this has already been addressed. Logic here should be `!dynamic_cast<...>` and not `dynamic_cast<...>`

**Review by mijovic — APPROVED** [review]

**Pull request merged**